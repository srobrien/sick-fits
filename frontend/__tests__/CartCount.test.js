import CartCounter from '../components/CartCounter';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('<CartCounter />', () => {
  it('renders', () => {
    shallow(<CartCounter count={10} />);
  });
  it('matches the snapshot', () => {
    const wrapper = shallow(<CartCounter count={10} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('updates via props snapshot', () => {
    const wrapper = shallow(<CartCounter count={50} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
