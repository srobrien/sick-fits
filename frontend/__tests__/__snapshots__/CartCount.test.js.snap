// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCounter /> matches the snapshot 1`] = `
<CartCounter__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCounter__Dot>
        10
      </CartCounter__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCounter__AnimationStyles>
`;

exports[`<CartCounter /> updates via props snapshot 1`] = `
<CartCounter__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCounter__Dot>
        50
      </CartCounter__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCounter__AnimationStyles>
`;

exports[`<CartCounter /> updates via props snapshot 2`] = `
<CartCounter__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCounter__Dot>
        10
      </CartCounter__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCounter__AnimationStyles>
`;
